### PR TITLE
Add CI pipelines to the template repo

### LINKED ISSUE #3: Include .github CI/CD configuration
The first thing users will want to do with their new provider is open PRs (`make reviewable`?) and publish the images to the Upbound registry or others.

Make it easy to get started by wiring up common settings. The Terrajet guide should include any search/replace needed to get `.github` updated for their environment.

### PR DESCRIPTION
**Pull request by ulucinar**
Fixes #3 
This PR adds the `.github` folder to the template repo.

### CODE REVIEW

**Comment by ulucinar** [comment]
@turkenh, where should we document to run `./hack/prepare-ci.sh`?

File: hack/prepare-ci.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+PROVIDER_NAME="$1"
+find .github -type f -exec sed -i '' -e "s/{{Template}}/${PROVIDER_NAME}/g" {} \;
```
Comment (turkenh) [inline hack/prepare-ci.sh:20]: Not sure if we really need a separate script. 
Maybe we can just add this as yet another command in step 3 here: https://github.com/crossplane-contrib/terrajet/blob/main/docs/generating-a-provider.md#generate

Comment (ulucinar) [reply hack/prepare-ci.sh:20]: I'm fine both ways. We may add the command in the guide and if more is needed, collect them in a script afterwards.

**Review by turkenh — APPROVED** [review]

**Comment by ulucinar** [comment]
Thank you @turkenh!

**Pull request merged**